docket/resources/query.py

```diff
@@ -0,0 +1,681 @@
+                    q_fields['proto-name']= arg.lower()
+                    continue
+                elif arg.upper() in ("IGNORE-WEIGHT", ""):
+                    continue
+                else:
+                    raise BadRequest("Unrecognized clause in URI: {}".format(arg))
+            elif state == "HOST":
+                q_fields[state.lower()].append(arg)
+                state = None
+            elif state == "PORT":
+                q_fields[state.lower()].append(int(arg))
+                state = None
+            elif state == "PROTO":
+                q_fields[state.lower()].append(arg)
+                state = None
+            elif state == "NET":
+                # Read a network, then look for a mask
+                q_fields['nettmp']=arg
+                state = "NET1"
+            elif state == "NET1":
+                q_fields['net'].append("{}/{}".format(q_fields['nettmp'], arg))
+                state = None
+                del(q_fields['nettmp'])
+            elif state in ["BEFORE", "AFTER"]:
+                if parse_duration(arg):
+                    q_fields[state.lower()+'-ago'] = arg
+                else:
+                    try:
+                        q_fields[state.lower()] = inputs.datetime_from_iso8601(arg)
+                    except Exception as e:
+                        raise ValueError("Unrecognized time in URI: {}".format(arg))
+                state = None
+
+        if state != None:
+            raise BadRequest("Incomplete {} clause".format(state))
+        return q_fields
+
+    def post(self):
+        """ post - build a query based on submitted form or json
+                 - enqueue it for processing
+                 - return the ID, URL needed for a future get ID/MERGE_NAME.pcap request
+        """
+        Config.logger.info("query request: {}".format(_get_request_nfo()))
+        low = space_low()
+        if low:
+            Config.logger.error(low)
+            return low
+
+        try:
+            fields = self.parse_json()
+            q = Query(fields=fields)
+        except BadRequest as e:
+            return str(e)
+        except ValueError as e:
+            return "400 Bad Request:" + str(e)
+        return q.enqueue()
+
+    def get(self, path):
+        """ get  - build a query based on uri and enqueue it for processing
+                 - return the ID, URL needed for a future get ID/MERGE_NAME.pcap request
+        """
+        Config.logger.info("URI request: {}".format(_get_request_nfo()))
+        low = space_low()
+        if low:
+            Config.logger.error(low)
+            return low
+
+        try:
+            fields = self.parse_uri(path)
+            q = Query(fields=fields)
+        except BadRequest as e:
+            return str(e)
+        except ValueError as e:
+            return "400 Bad Request:" + str(e)
+        return q.enqueue()
+
+class RawRequest(Resource):
+    """ This class handles RAW Stenographer Query requests. """
```

Comment:
This isn't correct. `stenocurl` abuses HTTP by using a POST method where the query is plaintext as a "form key" with no value. It also passes limit constraints via headers as found [here](https://github.com/rocknsm/docket/blob/master/docket/resources/pcap.py#L386-L388)